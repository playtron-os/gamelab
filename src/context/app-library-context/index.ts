export * from "./app-library-context";
export { useAppLibraryContext } from "./use-app-library-context";
