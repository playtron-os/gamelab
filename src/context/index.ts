export * from "./confirmation-pop-up-context";
export * from "./app-library-context";
export * from "./loading-spinner-context";
