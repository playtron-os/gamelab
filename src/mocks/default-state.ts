export const DEFAULT_STATE_MOCK = {
  auth: {
    userId: "userId"
  }
};
