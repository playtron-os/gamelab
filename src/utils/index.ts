export * from "./auth";
export * from "./test-utils";
