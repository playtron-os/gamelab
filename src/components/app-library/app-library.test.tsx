import React from "react";
import * as context from "@/context/app-library-context";
import { useAppLibraryContext } from "@/context";
import { AppLibrary } from ".";
import { columnConfig } from "../../context/app-library-context/table-config";

import { renderWithAppLibraryContext } from "@/utils";
import { configureStore } from "@reduxjs/toolkit";
import {
  APP_LIBRARY_INITIAL_STATE,
  appLibrarySlice,
  authSlice,
  moveAppDialogSlice
} from "@/redux/modules";
import { useBoolean } from "ahooks";
import { STEAM_APP_INFORMATION_MOCKS } from "@/mocks/app";
import { DEFAULT_STATE_MOCK } from "@/mocks/default-state";

const MOCK_BOOLEAN_STATE_MANAGER = [
  false,
  {
    setTrue: jest.fn(),
    setFalse: jest.fn(),
    set: jest.fn(),
    toggle: jest.fn()
  }
] as ReturnType<typeof useBoolean>;

const TABLE_MOCK_DATA = [
  STEAM_APP_INFORMATION_MOCKS[0],
  STEAM_APP_INFORMATION_MOCKS[1]
];

const MOCK_LIBRARY_TABLE_DATA: context.AppLibraryContextProps = {
  columns: columnConfig,
  handlers: {
    refetchAllApps: jest.fn(),
    downloadApp: jest.fn(),
    uninstallApp: jest.fn(),
    pauseDownload: jest.fn(),
    cancelDownload: jest.fn(),
    handleAppDefaultAction: jest.fn(),
    openMoveAppDialog: jest.fn(),
    openBulkActionsMenu: jest.fn()
  },
  bulkActionsMenuStateManager: MOCK_BOOLEAN_STATE_MANAGER,
  selectedApps: new Set([]),
  setSelectedApps: jest.fn(),
  selectedIds: [],
  onSelectedIdsChange: jest.fn(),
  isEulaOpen: false,
  setIsEulaOpen: jest.fn(),
  eula: null,
  acceptEula: jest.fn()
};

const getMockStore = ({
  loading,
  loadingProgress,
  error
}: {
  loading?: boolean;
  loadingProgress?: any;
  error?: string;
} = {}) => {
  return configureStore({
    reducer: {
      appLibrary: appLibrarySlice.reducer,
      auth: authSlice.reducer,
      moveAppDialog: moveAppDialogSlice.reducer
    },
    preloadedState: {
      ...DEFAULT_STATE_MOCK,
      appLibrary: {
        ...APP_LIBRARY_INITIAL_STATE,
        apps: TABLE_MOCK_DATA,
        loading: loading ?? false,
        loadingProgress: loadingProgress ?? {},
        error: error || ""
      },
      moveAppDialog: {
        isMoveAppDialogOpen: false,
        appInfoArray: [],
        isMovingApp: false
      }
    }
  });
};

describe("<AppLibrary />", () => {
  const getBoundingClientRect = window.Element.prototype.getBoundingClientRect;

  jest.mock("@/context", () => ({
    ...jest.requireActual("@/context"),
    useAppLibraryContext: jest.fn().mockReturnValue(MOCK_LIBRARY_TABLE_DATA),
    AppLibraryContextProvider: ({ children }: any) => children,
    LoadingSpinnerContextProvider: ({ children }: any) => children,
    useLoadingSpinner: jest.fn(() => {
      return {
        isLoadingSpinnerActive: false,
        showLoadingSpinner: jest.fn(),
        hideLoadingSpinner: jest.fn()
      };
    })
  }));

  beforeEach(() => {
    window.Element.prototype.getBoundingClientRect = jest
      .fn()
      .mockReturnValue({ height: 1000, width: 1000 });
  });

  afterEach(() => {
    window.Element.prototype.getBoundingClientRect = getBoundingClientRect;
  });

  it("loading state snapshot", () => {
    const { asFragment, findByTestId } = renderWithAppLibraryContext(
      <AppLibrary />,
      undefined,
      getMockStore({ loading: true, loadingProgress: { steam: 30, gog: 56 } })
    );
    const loadingText = findByTestId("loading-text");
    expect(loadingText).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  it("error state snapshot", () => {
    const { asFragment, findByTestId } = renderWithAppLibraryContext(
      <AppLibrary />,
      undefined,
      getMockStore({ error: "Something went wrong" })
    );
    const errorText = findByTestId("error-text");
    expect(errorText).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  // TODO - fix tests
  it("loaded state snapshot", () => {
    const { asFragment } = renderWithAppLibraryContext(
      <AppLibrary />,
      undefined,
      getMockStore()
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders app data correctly", () => {
    const { getByText } = renderWithAppLibraryContext(
      <AppLibrary />,
      undefined,
      getMockStore()
    );

    expect(getByText(TABLE_MOCK_DATA[0].app.name)).toBeTruthy();
    expect(getByText(TABLE_MOCK_DATA[1].app.name)).toBeTruthy();
  });
});
