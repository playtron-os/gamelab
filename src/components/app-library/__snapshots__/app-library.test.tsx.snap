// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLibrary /> error state snapshot 1`] = `
<DocumentFragment>
  <span
    data-testid="error-text"
  >
    Something went wrong
  </span>
</DocumentFragment>
`;

exports[`<AppLibrary /> loaded state snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full px-4 select-none cursor-default text-sm xl:text-base"
  >
    <div>
      <div
        class="flex"
      >
        <div
          class="flex w-full flex-col"
        >
          <div
            aria-label="installedFilter"
            class="inline-flex"
            data-slot="base"
          >
            <div
              aria-label="installedFilter"
              aria-orientation="horizontal"
              class="flex h-fit items-center flex-nowrap overflow-x-scroll scrollbar-hide bg-transparent dark:bg-transparent gap-6 w-full relative rounded-none p-0 border-divider"
              data-slot="tabList"
              id="react-aria-:ra:"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="z-0 w-full py-1 flex group relative justify-center items-center cursor-pointer transition-opacity tap-highlight-transparent data-[disabled=true]:cursor-not-allowed data-[disabled=true]:opacity-30 data-[hover-unselected=true]:opacity-disabled outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 rounded-none max-w-fit px-3 h-12 text-lg"
                data-key="all"
                data-selected="true"
                data-slot="tab"
                id="react-aria-:ra:-tab-all"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="absolute z-0 h-[2px] bottom-0 shadow-[0_1px_0px_0_rgba(0,0,0,0.05)] rounded-none w-full bg-[--state-default]"
                  data-slot="cursor"
                />
                <div
                  class="relative z-10 whitespace-nowrap transition-colors text-default-500 group-data-[selected=true]:text-white"
                  data-slot="tabContent"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <span>
                      All Games
                    </span>
                  </div>
                </div>
              </button>
              <button
                aria-selected="false"
                class="z-0 w-full py-1 flex group relative justify-center items-center cursor-pointer transition-opacity tap-highlight-transparent data-[disabled=true]:cursor-not-allowed data-[disabled=true]:opacity-30 data-[hover-unselected=true]:opacity-disabled outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 rounded-none max-w-fit px-3 h-12 text-lg"
                data-key="installed"
                data-slot="tab"
                id="react-aria-:ra:-tab-installed"
                role="tab"
                tabindex="-1"
                type="button"
              >
                <div
                  class="relative z-10 whitespace-nowrap transition-colors text-default-500 group-data-[selected=true]:text-white"
                  data-slot="tabContent"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <span>
                      Installed
                    </span>
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="flex-col w-[380px] pt-2 pe-2"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="mb-1 font-bold"
            />
            <div
              class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
              data-filled="true"
              data-filled-within="true"
              data-slot="base"
            >
              <div
                class="h-full flex flex-col"
                data-slot="main-wrapper"
              >
                <div
                  class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-10 min-h-10 transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background outline-hover rounded-md border border-[--state-default-neutral] hover:border-[--state-selected]"
                  data-slot="input-wrapper"
                  style="cursor: text;"
                >
                  <div
                    class="inline-flex w-full items-center h-full box-border bg-transparent"
                    data-slot="inner-wrapper"
                  >
                    <input
                      aria-label="Search"
                      class="w-full font-normal !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small peer pr-6 rtl:pr-0 rtl:pl-6 group-data-[has-value=true]:text-default-foreground bg-transparent"
                      data-slot="input"
                      data-testid="text-input"
                      id="react-aria-:rf:"
                      placeholder="Search"
                      type="text"
                      value=""
                    />
                    <span
                      aria-label="clear input"
                      class="p-2 -m-2 z-10 hidden absolute end-3 start-auto appearance-none select-none opacity-0 hover:!opacity-100 cursor-pointer active:!opacity-70 rounded-full outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-large peer-data-[filled=true]:opacity-70 peer-data-[filled=true]:block transition-opacity motion-reduce:transition-none"
                      data-slot="clear-button"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        aria-hidden="true"
                        focusable="false"
                        height="1em"
                        role="presentation"
                        viewBox="0 0 24 24"
                        width="1em"
                      >
                        <path
                          d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <h2
          class="text-2xl my-3 px-2"
        >
          All Games (
        </h2>
      </div>
      <div
        class="h-[calc(100vh-125px)] w-full overflow-scroll p-2"
        data-testid="app-library-table"
      >
        <div
          class="overflow-auto"
          style="max-height: 100%;"
        >
          <div
            style="height: 220px;"
          >
            <table
              class="border-collapse table-fixed w-full"
            >
              <thead>
                <tr>
                  <th
                    class="text-left border-b-2 p-2 border-[--state-focused]"
                    colspan="1"
                    style="width: 100px;"
                  >
                    <div
                      class=""
                    />
                  </th>
                  <th
                    class="text-left border-b-2 p-2 border-[--state-focused]"
                    colspan="1"
                    style="width: 240px;"
                  >
                    <div
                      class="cursor-pointer select-none"
                    >
                      Title
                    </div>
                  </th>
                  <th
                    class="text-left border-b-2 p-2 border-[--state-focused]"
                    colspan="1"
                    style="width: 50px;"
                  >
                    <div
                      class="cursor-pointer select-none"
                    >
                      Size
                    </div>
                  </th>
                  <th
                    class="text-left border-b-2 p-2 border-[--state-focused]"
                    colspan="1"
                    style="width: 50px;"
                  >
                    <div
                      class="cursor-pointer select-none"
                    >
                      Provider
                    </div>
                  </th>
                  <th
                    class="text-left border-b-2 p-2 border-[--state-focused]"
                    colspan="1"
                    style="width: 70px;"
                  >
                    <div
                      class="cursor-pointer select-none"
                    >
                      Status
                    </div>
                  </th>
                  <th
                    class="text-left border-b-2 p-2 border-[--state-focused]"
                    colspan="1"
                    style="width: 50px;"
                  >
                    <div
                      class="cursor-pointer select-none"
                    >
                      Drive
                    </div>
                  </th>
                  <th
                    class="text-left border-b-2 p-2 border-[--state-focused]"
                    colspan="1"
                    style="width: 32px;"
                  >
                    <div
                      class="cursor-pointer select-none"
                    />
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  class="bg-[--fill-default]"
                  style="height: 60px; transform: translateY(0px);"
                >
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <div
                      class="flex justify-center mx-2 w-[100px] h-[48px]"
                    >
                      <img
                        alt=""
                        class="h-full w-full max-h-[48px] max-w-[100px] object-cover"
                        loading="lazy"
                        src="0"
                      />
                    </div>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <div
                      class="my-3 px-2 overflow-ellipsis text-nowrap font-bold"
                    >
                      Apex Legends
                    </div>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <div
                      class="px-3 py-3 text-nowrap text-sm"
                    >
                      100 B
                    </div>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <span
                      class="px-3 my-3 text-nowrap text-sm overflow-ellipsis"
                    >
                      Steam, Epic Games
                    </span>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <span
                      class="p-3 text-nowrap text-sm"
                    >
                      Ready
                    </span>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <div
                      class="p-3 text-nowrap text-sm"
                    >
                      C:
                    </div>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased inline-block"
                      data-slot="trigger"
                      id="react-aria-:rk:"
                      type="button"
                    >
                      <svg
                        class="cursor-pointer"
                        data-testid="app-action-cell-0"
                        fill="#FFFFFF"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 8C10.8954 8 10 7.10457 10 6C10 4.89543 10.8954 4 12 4C13.1046 4 14 4.89543 14 6C14 7.10457 13.1046 8 12 8Z"
                        />
                        <path
                          d="M12 14C10.8954 14 10 13.1046 10 12C10 10.8954 10.8954 10 12 10C13.1046 10 14 10.8954 14 12C14 13.1046 13.1046 14 12 14Z"
                        />
                        <path
                          d="M10 18C10 19.1046 10.8954 20 12 20C13.1046 20 14 19.1046 14 18C14 16.8954 13.1046 16 12 16C10.8954 16 10 16.8954 10 18Z"
                        />
                      </svg>
                    </div>
                  </td>
                </tr>
                <tr
                  class="bg-[--fill-subtle]"
                  style="height: 60px; transform: translateY(0px);"
                >
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <div
                      class="flex justify-center mx-2 w-[100px] h-[48px]"
                    >
                      <img
                        alt=""
                        class="h-full w-full max-h-[48px] max-w-[100px] object-cover"
                        loading="lazy"
                        src="0"
                      />
                    </div>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <div
                      class="my-3 px-2 overflow-ellipsis text-nowrap font-bold"
                    >
                      Counter-strike: Global Offensive
                    </div>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <div
                      class="px-3 py-3 text-nowrap text-sm"
                    >
                      100 B
                    </div>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <span
                      class="px-3 my-3 text-nowrap text-sm overflow-ellipsis"
                    >
                      Steam, Epic Games
                    </span>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <span
                      class="p-3 text-nowrap text-sm"
                    >
                      Ready
                    </span>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <div
                      class="p-3 text-nowrap text-sm"
                    >
                      C:
                    </div>
                  </td>
                  <td
                    class="overflow-hidden border-b-8 border-[--fill-subtler]"
                    style="width: auto; min-width: unset; position: relative; height: 60px;"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased inline-block"
                      data-slot="trigger"
                      id="react-aria-:rn:"
                      type="button"
                    >
                      <svg
                        class="cursor-pointer"
                        data-testid="app-action-cell-1"
                        fill="#FFFFFF"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 8C10.8954 8 10 7.10457 10 6C10 4.89543 10.8954 4 12 4C13.1046 4 14 4.89543 14 6C14 7.10457 13.1046 8 12 8Z"
                        />
                        <path
                          d="M12 14C10.8954 14 10 13.1046 10 12C10 10.8954 10.8954 10 12 10C13.1046 10 14 10.8954 14 12C14 13.1046 13.1046 14 12 14Z"
                        />
                        <path
                          d="M10 18C10 19.1046 10.8954 20 12 20C13.1046 20 14 19.1046 14 18C14 16.8954 13.1046 16 12 16C10.8954 16 10 16.8954 10 18Z"
                        />
                      </svg>
                    </div>
                  </td>
                </tr>
              </tbody>
              <tfoot>
                <tr>
                  <th />
                  <th />
                  <th />
                  <th />
                  <th />
                  <th />
                  <th />
                </tr>
              </tfoot>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<AppLibrary /> loading state snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full px-4 select-none cursor-default text-sm xl:text-base"
  >
    <div>
      <div
        class="flex"
      >
        <div
          class="flex w-full flex-col"
        >
          <div
            aria-label="installedFilter"
            class="inline-flex"
            data-slot="base"
          >
            <div
              aria-label="installedFilter"
              aria-orientation="horizontal"
              class="flex h-fit items-center flex-nowrap overflow-x-scroll scrollbar-hide bg-transparent dark:bg-transparent gap-6 w-full relative rounded-none p-0 border-divider"
              data-slot="tabList"
              id="react-aria-:r0:"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="z-0 w-full py-1 flex group relative justify-center items-center cursor-pointer transition-opacity tap-highlight-transparent data-[disabled=true]:cursor-not-allowed data-[disabled=true]:opacity-30 data-[hover-unselected=true]:opacity-disabled outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 rounded-none max-w-fit px-3 h-12 text-lg"
                data-key="all"
                data-selected="true"
                data-slot="tab"
                id="react-aria-:r0:-tab-all"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="absolute z-0 h-[2px] bottom-0 shadow-[0_1px_0px_0_rgba(0,0,0,0.05)] rounded-none w-full bg-[--state-default]"
                  data-slot="cursor"
                />
                <div
                  class="relative z-10 whitespace-nowrap transition-colors text-default-500 group-data-[selected=true]:text-white"
                  data-slot="tabContent"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <span>
                      All Games
                    </span>
                  </div>
                </div>
              </button>
              <button
                aria-selected="false"
                class="z-0 w-full py-1 flex group relative justify-center items-center cursor-pointer transition-opacity tap-highlight-transparent data-[disabled=true]:cursor-not-allowed data-[disabled=true]:opacity-30 data-[hover-unselected=true]:opacity-disabled outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 rounded-none max-w-fit px-3 h-12 text-lg"
                data-key="installed"
                data-slot="tab"
                id="react-aria-:r0:-tab-installed"
                role="tab"
                tabindex="-1"
                type="button"
              >
                <div
                  class="relative z-10 whitespace-nowrap transition-colors text-default-500 group-data-[selected=true]:text-white"
                  data-slot="tabContent"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <span>
                      Installed
                    </span>
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="flex-col w-[380px] pt-2 pe-2"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="mb-1 font-bold"
            />
            <div
              class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
              data-filled="true"
              data-filled-within="true"
              data-slot="base"
            >
              <div
                class="h-full flex flex-col"
                data-slot="main-wrapper"
              >
                <div
                  class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-10 min-h-10 transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background outline-hover rounded-md border border-[--state-default-neutral] hover:border-[--state-selected]"
                  data-slot="input-wrapper"
                  style="cursor: text;"
                >
                  <div
                    class="inline-flex w-full items-center h-full box-border bg-transparent"
                    data-slot="inner-wrapper"
                  >
                    <input
                      aria-label="Search"
                      class="w-full font-normal !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small peer pr-6 rtl:pr-0 rtl:pl-6 group-data-[has-value=true]:text-default-foreground bg-transparent"
                      data-slot="input"
                      data-testid="text-input"
                      id="react-aria-:r5:"
                      placeholder="Search"
                      type="text"
                      value=""
                    />
                    <span
                      aria-label="clear input"
                      class="p-2 -m-2 z-10 hidden absolute end-3 start-auto appearance-none select-none opacity-0 hover:!opacity-100 cursor-pointer active:!opacity-70 rounded-full outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-large peer-data-[filled=true]:opacity-70 peer-data-[filled=true]:block transition-opacity motion-reduce:transition-none"
                      data-slot="clear-button"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        aria-hidden="true"
                        focusable="false"
                        height="1em"
                        role="presentation"
                        viewBox="0 0 24 24"
                        width="1em"
                      >
                        <path
                          d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <h2
          class="text-2xl my-3 px-2"
        >
          All Games (
        </h2>
      </div>
      <div
        class="absolute top-0 bottom-0 right-0 left-0 h-screen w-screen bg-black/40 z-20"
      >
        <div
          class="flex width-full h-screen w-screen justify-center items-center"
        >
          <div
            data-testid="progress-spinner"
          >
            <svg
              fill="none"
              height="48"
              viewBox="0 0 26 26"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.464 8.464C8.27647 8.65147 8.02216 8.75679 7.757 8.75679C7.49183 8.75679 7.23753 8.65147 7.05 8.464L4.93 6.344C4.74236 6.15649 4.63689 5.90212 4.6368 5.63685C4.6367 5.37158 4.74199 5.11714 4.9295 4.9295C5.11701 4.74186 5.37137 4.63639 5.63664 4.6363C5.90191 4.6362 6.15636 4.74149 6.344 4.929L8.464 7.05C8.65147 7.23753 8.75678 7.49184 8.75678 7.757C8.75678 8.02216 8.65147 8.27647 8.464 8.464Z"
                data-spinner-id="1"
                fill="#FFFFFF"
              />
              <path
                d="M7 12C7 12.2652 6.89464 12.5196 6.70711 12.7071C6.51957 12.8946 6.26522 13 6 13H3C2.73478 13 2.48043 12.8946 2.29289 12.7071C2.10536 12.5196 2 12.2652 2 12C2 11.7348 2.10536 11.4804 2.29289 11.2929C2.48043 11.1054 2.73478 11 3 11H6C6.26522 11 6.51957 11.1054 6.70711 11.2929C6.89464 11.4804 7 11.7348 7 12Z"
                data-spinner-id="2"
                fill="#FFFFFF"
              />
              <path
                d="M4.92999 19.071C4.74252 18.8835 4.63721 18.6292 4.63721 18.364C4.63721 18.0988 4.74252 17.8445 4.92999 17.657L7.05099 15.536C7.14324 15.4405 7.25358 15.3643 7.37559 15.3119C7.49759 15.2595 7.62881 15.2319 7.76159 15.2307C7.89437 15.2296 8.02605 15.2549 8.14895 15.3052C8.27184 15.3555 8.3835 15.4297 8.47739 15.5236C8.57128 15.6175 8.64553 15.7291 8.69582 15.852C8.7461 15.9749 8.7714 16.1066 8.77024 16.2394C8.76909 16.3722 8.7415 16.5034 8.68909 16.6254C8.63669 16.7474 8.5605 16.8578 8.46499 16.95L6.34499 19.071C6.25212 19.164 6.14183 19.2377 6.02043 19.2881C5.89904 19.3384 5.76891 19.3643 5.63749 19.3643C5.50608 19.3643 5.37595 19.3384 5.25455 19.2881C5.13315 19.2377 5.02287 19.164 4.92999 19.071Z"
                data-spinner-id="3"
                fill="#FFFFFF"
              />
              <path
                d="M12 17C12.2652 17 12.5196 17.1054 12.7071 17.2929C12.8946 17.4804 13 17.7348 13 18V21C13 21.2652 12.8946 21.5196 12.7071 21.7071C12.5196 21.8946 12.2652 22 12 22C11.7348 22 11.4804 21.8946 11.2929 21.7071C11.1054 21.5196 11 21.2652 11 21V18C11 17.7348 11.1054 17.4804 11.2929 17.2929C11.4804 17.1054 11.7348 17 12 17Z"
                data-spinner-id="4"
                fill="#FFFFFF"
              />
              <path
                d="M19.071 19.071C18.8835 19.2585 18.6292 19.3638 18.364 19.3638C18.0988 19.3638 17.8445 19.2585 17.657 19.071L15.536 16.95C15.3538 16.7614 15.253 16.5088 15.2553 16.2466C15.2576 15.9844 15.3628 15.7336 15.5482 15.5482C15.7336 15.3628 15.9844 15.2576 16.2466 15.2553C16.5088 15.253 16.7614 15.3538 16.95 15.536L19.071 17.656C19.164 17.7489 19.2377 17.8592 19.2881 17.9806C19.3384 18.102 19.3643 18.2321 19.3643 18.3635C19.3643 18.4949 19.3384 18.625 19.2881 18.7464C19.2377 18.8678 19.164 18.9781 19.071 19.071Z"
                data-spinner-id="5"
                fill="#FFFFFF"
              />
              <path
                d="M22 12C22 12.2652 21.8946 12.5196 21.7071 12.7071C21.5196 12.8946 21.2652 13 21 13H18C17.7348 13 17.4804 12.8946 17.2929 12.7071C17.1054 12.5196 17 12.2652 17 12C17 11.7348 17.1054 11.4804 17.2929 11.2929C17.4804 11.1054 17.7348 11 18 11H21C21.2652 11 21.5196 11.1054 21.7071 11.2929C21.8946 11.4804 22 11.7348 22 12Z"
                data-spinner-id="6"
                fill="#FFFFFF"
              />
              <path
                d="M15.536 8.464C15.3485 8.27647 15.2432 8.02216 15.2432 7.757C15.2432 7.49184 15.3485 7.23753 15.536 7.05L17.656 4.929C17.8435 4.74136 18.0979 4.63589 18.3631 4.6358C18.6284 4.6357 18.8829 4.74099 19.0705 4.9285C19.2581 5.11601 19.3636 5.37038 19.3637 5.63565C19.3638 5.90092 19.2585 6.15536 19.071 6.343L16.95 8.464C16.7625 8.65147 16.5082 8.75679 16.243 8.75679C15.9778 8.75679 15.7235 8.65147 15.536 8.464Z"
                data-spinner-id="7"
                fill="#FFFFFF"
              />
              <path
                d="M12 2C12.2652 2 12.5196 2.10536 12.7071 2.29289C12.8946 2.48043 13 2.73478 13 3V6C13 6.26522 12.8946 6.51957 12.7071 6.70711C12.5196 6.89464 12.2652 7 12 7C11.7348 7 11.4804 6.89464 11.2929 6.70711C11.1054 6.51957 11 6.26522 11 6V3C11 2.73478 11.1054 2.48043 11.2929 2.29289C11.4804 2.10536 11.7348 2 12 2Z"
                data-spinner-id="8"
                fill="#FFFFFF"
                opacity="0.4"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
